src/librustc/middle/ty.rs

```diff
@@ -5459,6 +5465,11 @@ impl<'tcx> ctxt<'tcx> {
                         &format!("consider boxing your closure and/or \
                                   using it as a trait object"));
                 }
+            },
+            terr_ty_param_default_mismatch(expected) => {
+                self.sess.span_note(sp,
+                    &format!("found conflicting defaults {:?} {:?}",
```

Comment:
Yeah my perfect error message would underline the two defaults and show exactly where the conflicting, way delaying the work until I got some feedback. I think it would be possible to record both a default and a span in the defaults table, and use that for reporting.